File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int i=0,j=0,k=0;
        int n1=nums1.length;
        int n2=nums2.length;
        int[] merged=new int[n1+n2];
        int m=merged.length;
        while(i<n1 && j<n2){
            if(nums1[i]<nums2[j]){
                merged[k++]=nums1[i++];
            }else{
                merged[k++]=nums2[j++];
            }
        }  
        while(i<n1){
            merged[k++]=nums1[i++];
        }
        while(j<n2){
            merged[k++]=nums2[j++];
        }
        if(m%2==0){
            return (merged[m/2]+merged[(m-1)/2])/2.0;
        }else{
            return merged[m/2];
        }
        
        
    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cs
public class Solution {
    public double FindMedianSortedArrays(int[] nums1, int[] nums2) {
        int n1=nums1.Length;
        int n2=nums2.Length;
        int n=n1+n2;
        int idx2=n/2;
        int idx1=idx2-1;
        int i=0,j=0;
        int count=0;
        int el1=0,el2=0;
        while(i<n1 && j<n2){
           if(nums1[i]>nums2[j]){
             if(count==idx1)el1=nums2[j];
             if(count==idx2)el2=nums2[j];
             j++;
             count++;
           }else{
             if(count==idx1)el1=nums1[i];
             if(count==idx2)el2=nums1[i];
             i++;
             count++;
           }
        }
        while(i<n1){
            if(count==idx1)el1=nums1[i];
            if(count==idx2)el2=nums1[i];
            i++;
            count++;
        }
        while(j<n2){
            if(count==idx1)el1=nums2[j];
            if(count==idx2)el2=nums2[j];
            j++;
            count++;
        }
        if(n%2==1){
            return el2;
        }else{
            return (double)(el1+el2)/2;
        }
    }
}